### ISSUE TITLE
can I use wlan0 instead of eth0?

### ISSUE THREAD

**Issue by mirkomancin** [cetic/6lbr#9]
I need to make a border router using the wifi (I have a usb dongle) and let me know if you can use the examples that we are here on github. How do I set 6lbr on wlan0?

**Comment by Suretronic** [comment]
Sorry to say that so far after trying the above and many other options I've failed to get a wireless lan working in Smart Bridge mode.

**Comment by shantanoo-desai** [comment]
This may also be dependent on the WiFi Dongle you may be using in most common cases the __Edimax 7811UN__ should work fine. There has definitely been some problems with using specific type of WiFi USB dongles when using in Ad-Hoc mode of operations if that is what you wish to achieve through the RPi and 6LBR service. Currently I am working on a different concept with RPi and WiFi and if you need some insight on basic understanding here is the link to a [StackOverflow Query](http://stackoverflow.com/questions/34157743/raspberry-pi-and-edimax-7811un-incompatibility-in-ad-hoc-mode/34458131#34458131) I posted and resolved.

This is a bit off topic but maybe there might be hardware compatibility issues in general that might be causing problems.
I hope it helps in a way.

**Comment by Suretronic** [comment]
The hardware suggestion could be useful thanks Shantanoo. Quick check reveals
that the Edimax 7811UN uses the RTL8188CUS chip, whereas the dongle that I've
been testing is based on the RTL8192CU chip and these have different drivers. 
So I will source an RTL8188CUS based device and try that. Best wishes, Ron.

**Comment by rohitsivakumar** [comment]
I tried running the 6lbr on BananaPI-M1+ board running a RASPBIAN OS. I was able to run the 6lbr service on it but I don't think the bridge is connected between wlan0 and the wsn network. The BananaPI-M1+ comes with a AP6120 wifi chipset. 

Any ideas where I could look at. 
Here are the details of my status messages and if someone can comment on why it isn't create a bridge, there is no bbbb::100 though nvm.dat prints it.

**6lbr.log:**
```
----------------------------------
Tue Apr 12 11:05:21 UTC 2016 : Starting 6LBR                                    
/usr/lib/6lbr/bin/cetic_6lbr_smart_bridge -c /etc/6lbr/nvm.dat -s /dev/ttyACM0 -
t wlan0 -r -B 115200 -U /usr/lib/6lbr/6lbr-ifup -D /usr/lib/6lbr/6lbr-ifdown -w 
/usr/lib/6lbr/www -L 3  -W /var/log/6lbr.timestamp -P 60 -C /var/log/6lbr.ip  -m
 /usr/lib/6lbr/plugins                                                          
Contiki-6lbr-1.3.3 started with IPV6, RPL                                       
Rime started with address 1.2.3.4.5.6.7.8                                       
MAC CSMA RDC br-rdc NETWORK sicslowpan                                          
Log level : 30                                                                  
Log services : ffffffff                                                         
2016-03-12 11:05:21.927638: INFO: ETH: 6LBR watchdog started (interval: 60)     
2016-03-12 11:05:21.928622: INFO: 6LBR: Starting 6LBR version 1.3.3 (Contiki-6lb
r-1.3.3)                                                                        
2016-03-12 11:05:21.928669: INFO: NVM: Opening nvm file '/etc/6lbr/nvm.dat'     
2016-03-12 11:05:21.928770: INFO: NVM: NVM Magic : 2009                         
2016-03-12 11:05:21.928794: INFO: NVM: NVM Version : 1                          
2016-03-12 11:05:21.928811: INFO: SCMD: Started br-cmd process                  
2016-03-12 11:05:21.928825: INFO: NODECFG: No node_config.conf file specified   
2016-03-12 11:05:21.928855: INFO: ETH: RAW/TAP init                             
2016-03-12 11:05:21.930227: INFO: SLIP: SLIP started on /dev/ttyACM0            
2016-03-12 11:05:21.944712: INFO: TAP: opened device /dev/wlan0                 
2016-03-12 11:05:21.945000: INFO: TAP: Running 6lbr-ifup script '/usr/lib/6lbr/6
lbr-ifup'                                                                       
wlan0     Link encap:Ethernet  HWaddr 98:3b:16:c7:d4:ba                         
          inet addr:192.168.1.160  Bcast:192.168.1.255  Mask:255.255.255.0      
          inet6 addr: fe80::9a3b:16ff:fec7:d4ba/64 Scope:Link                   
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1                    
          RX packets:46155 errors:0 dropped:0 overruns:0 frame:0                
          TX packets:18952 errors:0 dropped:0 overruns:0 carrier:0              
          collisions:0 txqueuelen:1000                                          
          RX bytes:31027285 (29.5 MiB)  TX bytes:1226471 (1.1 MiB)              
                                                                                
2016-03-12 11:05:22.121806: INFO: TAP: Eth MAC address : 98:3b:16:c7:d4:ba      
2016-03-12 11:05:22.121981: INFO: BR-RDC: Reset SLIP Radio                      
2016-03-12 11:05:22.122058: INFO: BR-RDC: Fetching MAC address                  
2016-03-12 11:05:22.155227: INFO: SLIP: cc2531: Reboot requested                
2016-03-12 11:05:22.157076: INFO: SLIP: cc2531: Reverting to h/w MAC but not reb
ooting                                                                          
2016-03-12 11:05:22.186116: INFO: BR-RDC: Got MAC: 00:12:4b:00:07:68:ce:5d      
2016-03-12 11:05:23.122594: INFO: 6LBR: Tentative local IPv6 address fe80::212:4
b00:768:ce5d                                                                    
2016-03-12 11:05:23.123025: INFO: NVM: Opening nvm file '/etc/6lbr/nvm.dat'     
2016-03-12 11:05:23.123709: INFO: 6LBR: Configured as DODAG Root                
2016-03-12 11:05:23.123857: INFO: 6LBR: Starting as SMART-BRIDGE                
2016-03-12 11:05:23.125047: INFO: UDPS: UDP server started                      
2016-03-12 11:05:23.125543: INFO: 6LBR: CETIC 6LBR Started                      
2016-03-12 11:05:23.125798: INFO: SLIP: cc2531: Setting RF channel 0x19 
-------------------------------------------------------------------------
```
**6lbr.ip:**
`::`
--------------------------------------------------------------------------
**6lbr.conf:**
```
#MODE=ROUTER                                                                    
MODE=SMART-BRIDGE                                                               
#MODE=RPL-RELAY                                                                 
#MODE=FULL-TRANSPARENT-BRIDGE                                                   
#MODE=NDP-ROUTER                                                                
#MODE=6LR                                                                       
#MODE=RPL-ROOT                                                                  
                                                                                
RAW_ETH=1                                                                       
BRIDGE=0                                                                        
DEV_BRIDGE=br0                                                                  
DEV_TAP=tap0                                                                    
DEV_ETH=wlan0                                                                   
RAW_ETH_FCS=0                                                                   
                                                                                
DEV_RADIO=/dev/ttyACM0                                                          
BAUDRATE=115200                                                                 
                                                                                
LOG_LEVEL=3 #INFO and above only
```
**nvm.dat print:
**Reading nvm file '/etc/6lbr/nvm.dat'                                            
Channel : 25                                                                    
                                                                                
WSN network prefix : aaaa::                                                     
WSN network prefix length : 64                                                  
WSN IP address : aaaa::100                                                      
WSN accept RA : True                                                            
WSN IP address autoconf : True                                                  
                                                                                
Eth network prefix : bbbb::                                                     
Eth network prefix length : 6 address : bbbb::100                               
Eth default router : ::                                                         
Eth IP address autoconf : False                                                 
                                                                                
Local address rewrite : True                                                    
Smart Multi BR : False                                                          
                                                                                
RA daemon : True                                                                
RA router lifetime : 0                                                          
RA maximum interval : 600                                                       
RA minimum interval : 200                                                       
RA minimum delay : 3                                                            
RA PIO enabled : Tbled : True                                                   
RA RIO lifetime : 1800                                                          
                                                                                
RPL instance ID : 30                                                            
RPL Preference : 0                                                              
RPL version ID :  : 250                                                         
RPL DIO interval doubling : 8                                                   
RPL DIO minimum interval : 12                                                   
RPL DIO redundancy : 10                                                         
RPL default lifetime : 30                                                       
RPL minimum rank increment : 256                                                
RPL lifetime unit : 256                                                         
                                                                                
Webserver configuration page disabled : False

**Comment by rohitsivakumar** [comment]
One additional observation I made: When I changed the mode from smart-bridge to Router in the 6lbr.conf file and restart the 6lbr service and check the 6lbr.log I see a ipv6 address assigned:
```
MODE=ROUTER                                                                     
#MODE=SMART-BRIDGE 
```
**Here is the 6lbr.ip snap:**
```
pi@bananapi:/etc/6lbr$ cat /var/log/6lbr.ip                                     
bbbb::100
```
**6lbr.log now shows:**
```
********************************************************************************
Tue Apr 12 11:26:21 UTC 2016 : Starting 6LBR                                    
/usr/lib/6lbr/bin/cetic_6lbr_router -c /etc/6lbr/nvm.dat -s /dev/ttyACM0 -t wlan
0 -r -B 115200 -U /usr/lib/6lbr/6lbr-ifup -D /usr/lib/6lbr/6lbr-ifdown -w /usr/l
ib/6lbr/www -L 3  -W /var/log/6lbr.timestamp -P 60 -C /var/log/6lbr.ip  -m /usr/
lib/6lbr/plugins                                                                
Contiki-6lbr-1.3.3 started with IPV6, RPL                                       
Rime started with address 1.2.3.4.5.6.7.8                                       
MAC CSMA RDC br-rdc NETWORK sicslowpan                                          
Log level : 30                                                                  
Log services : ffffffff                                                         
2016-03-12 11:26:21.243197: INFO: ETH: 6LBR watchdog started (interval: 60)     
2016-03-12 11:26:21.244270: INFO: 6LBR: Starting 6LBR version 1.3.3 (Contiki-6lb
r-1.3.3)                                                                        
2016-03-12 11:26:21.244316: INFO: NVM: Opening nvm file '/etc/6lbr/nvm.dat'     
2016-03-12 11:26:21.244418: INFO: NVM: NVM Magic : 2009                         
2016-03-12 11:26:21.244445: INFO: NVM: NVM Version : 1                          
2016-03-12 11:26:21.244462: INFO: SCMD: Started br-cmd process                  
2016-03-12 11:26:21.244486: INFO: NODECFG: No node_config.conf file specified   
2016-03-12 11:26:21.244515: INFO: ETH: RAW/TAP init                             
2016-03-12 11:26:21.245826: INFO: SLIP: SLIP started on /dev/ttyACM0            
2016-03-12 11:26:21.260425: INFO: TAP: opened device /dev/wlan0                 
2016-03-12 11:26:21.260672: INFO: TAP: Running 6lbr-ifup script '/usr/lib/6lbr/6
lbr-ifup'                                                                       
wlan0     Link encap:Ethernet  HWaddr 98:3b:16:c7:d4:ba                         
          inet addr:192.168.1.160  Bcast:192.168.1.255  Mask:255.255.255.0      
          inet6 addr: fe80::9a3b:16ff:fec7:d4ba/64 Scope:Link                   
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1                    
          RX packets:49020 errors:0 dropped:0 overruns:0 frame:0                
          TX packets:19032 errors:0 dropped:0 overruns:0 carrier:0              
          collisions:0 txqueuelen:1000                                          
          RX bytes:31289409 (29.8 MiB)  TX bytes:1233920 (1.1 MiB)              
                                                                                
2016-03-12 11:26:21.436143: INFO: TAP: Eth MAC address : 98:3b:16:c7:d4:ba      
2016-03-12 11:26:21.436322: INFO: BR-RDC: Reset SLIP Radio                      
2016-03-12 11:26:21.436400: INFO: BR-RDC: Fetching MAC address                  
2016-03-12 11:26:21.469715: INFO: SLIP: cc2531: Reboot requested                
2016-03-12 11:26:21.470745: INFO: SLIP: cc2531: Reverting to h/w MAC but not reb
ooting                                                                          
2016-03-12 11:26:21.499750: INFO: BR-RDC: Got MAC: 00:12:4b:00:07:68:ce:5d      
2016-03-12 11:26:22.436515: INFO: 6LBR: Tentative local IPv6 address fe80::212:4
b00:768:ce5d                                                                    
2016-03-12 11:26:22.436948: INFO: 6LBR: Tentative global IPv6 address (WSN) aaaa
::212:4b00:768:ce5d                                                             
2016-03-12 11:26:22.437220: INFO: 6LBR: Tentative global IPv6 address (ETH) bbbb
::100                                                                           
2016-03-12 11:26:22.437380: INFO: 6LBR: RA Daemon enabled                       
2016-03-12 11:26:22.437485: INFO: NVM: Opening nvm file '/etc/6lbr/nvm.dat'     
2016-03-12 11:26:22.438168: INFO: 6LBR: Configured as DODAG Root                
2016-03-12 11:26:22.438313: INFO: 6LBR: Starting as RPL ROUTER                  
2016-03-12 11:26:22.439546: INFO: UDPS: UDP server started                      
2016-03-12 11:26:22.440023: INFO: 6LBR: CETIC 6LBR Started                      
2016-03-12 11:26:22.440249: INFO: SLIP: cc2531: Setting RF channel 0x19
```

**6lbr service status:**
```
pi@bananapi:/etc/6lbr$ sudo /etc/init.d/6lbr status                             
6lbr is running. 
```

**ifconfig details:**
```
pi@bananapi:/etc/6lbr$ ifconfig
eth0      Link encap:Ethernet  HWaddr 02:4e:06:c2:d9:95  
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:12 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:0 (0.0 B)  TX bytes:1740 (1.6 KiB)
          Iddress:0xc000 

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:16436  Metric:1
          RX packets:8 errors:0 dropped:0 overruns:0 frame:0
          TX packets:8 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:1104 (1.0 KiB)  TX bytes:1104 (1.0 KiB)

p2p0      Link encap:Ethernet  HWaddr 9a:3b:16:c7:d4:ba  
          inet6 addr: fe80::983b:16ff:fets:0 errors:0 dropped:0 overruns:0 frame
:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)                                
                                                                                
tunl0     Link encap:IPIP Tunnel  HWaddr                                        
          UP RUNNING NOARP  MTU:1480  Metric:1                                  
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0                    
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0                  
          collisit  HWaddr 98:3b:16:c7:d4:ba                                    
          inet addr:192.168.1.160  Bcast:192.168.1.255  Mask:255.255.255.0      
          inet6 addr: fe80::9a3b:16ff:fec7:d4ba/64 Scope:Link                   
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1                    
          RX packets:50244 errors:0 dropped:0 overruns:0 frame:0                
          TX packets:19053 errors:0 dropped:0 overruns:0 carrier:0              
          collisions:0 txqueuelen:1000                                          
          RX bytes:31404595 (29.9 MiB)  TX bytes:12358

wlan0     Link encap:Ethernet  HWaddr 98:3b:16:c7:d4:ba                         
          inet addr:192.168.1.160  Bcast:192.168.1.255  Mask:255.255.255.0      
          inet6 addr: fe80::9a3b:16ff:fec7:d4ba/64 Scope:Link                   
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1                    
          RX packets:50300 errors:0 dropped:0 overruns:0 frame:0                
          TX packets:19053 errors:0 dropped:0 overruns:0 carrier:0              
          collisions:0 txqueuelen:1000                                          
          RX bytes:31409866 (29.9 MiB)  TX bytes:1235879 (1.1 MiB)
```

**BananaPI-M1+ OS details:**
```
pi@bananapi:/etc/6lbr$ uname -r                                                 
3.4.108                                                                         
pi@bananapi:/etc/6lbr$ uname -a                                                 
Linux bananapi 3.4.108 #2 SMP PREEMPT Sat Aug 1 09:31:47 PDT 2015 armv7l GNU/Lin
ux               
```

**Comment by laurentderu** [comment]
You have to configure the interface of 6lbr to Bridging, currently you have it set up as Raw Ethernet :

    RAW_ETH=1                                                                       
    BRIDGE=0   

see https://github.com/cetic/6lbr/wiki/6LBR-Interface-Configuration

In order to have a bridge between 6LBR and your WiFi interface, your WiFi chipset must support promiscuous mode or it won't work.

**Comment by rohitsivakumar** [comment]
Thanks Laurentderu for sharing the info. Out of the three suggested modes (Raw ethernet, Bridge, Router mode) which one should be the best one to choose? If I go with the bridge mode, there is small statement stating "As you see, the Ethernet interface is no longer configured, it has been replaced by br0. t is possible to create the Bridge when 6LBR starts up, but this has as consequence that you will lose all your existing connections when the Ethernet interface join the bridge".

So I understand I will lose my ethernet connectivity but what actually does this statement mean in terms of establishing a working network between WSN and WLAN connectivity on the Linux host?

**Comment by laurentderu** [comment]
Your connection will be reset if (and only if) you ask 6LBR to create the bridge interface (**CREATE_BRIDGE=1**) and make the Ethernet interface join that bridge.

The proper way is to let the system create the bridge interface and add the Ethernet interface to it as explained in the Wiki page. That way, the 6LBR can join and leave the bridge interface without impacting any existing connections.

By connection, I mean, for example, your ssh connection towards your host.

**Comment by rohitsivakumar** [comment]
Ok, i guess this above mentioned approach works only on eth0. I tried this with wlan0. Doesn't seem to be kicking off!

Here's my 6lbr.conf configuration
```
#This file contains a default configuration for Raspberry PI platform using
#a Telos SLIP Radio
#The full list of parameters and their meaning can be found in 6lbr.conf.example

#MODE=ROUTER
MODE=SMART-BRIDGE
#MODE=RPL-RELAY
#MODE=FULL-TRANSPARENT-BRIDGE
#MODE=NDP-ROUTER
#MODE=6LR
#MODE=RPL-ROOT

RAW_ETH=0  
BRIDGE=1
CREATE_BRIDGE=1
DEV_BRIDGE=br0
DEV_TAP=tap0
DEV_ETH=wlan0
RAW_ETH_FCS=0

DEV_RADIO=/dev/ttyACM0
BAUDRATE=115200

LOG_LEVEL=3 #INFO and above only
```

My interfaces configuration:
```
auto lo

iface lo inet loopback
iface eth0 inet dhcp

allow-hotplug wlan0
#iface wlan0 inet manual
iface wlan0 inet static
address 0.0.0.0

auto br0
iface br0 inet dhcp
        bridge_ports wlan0
        bridge_stp off
        up echo 0 > sys/devices/virtual/br0/bridge/multicast_snooping
        post-up ip link set br0 address 'ip link show wlan0 | grep ether | awk  
wpa-roam /etc/wpa_supplicant/wpa_supplicant.conf
iface default inet dhcp
```
```
pi@bananapi:/etc/network$ sudo /etc/init.d/6lbr status
6lbr is running.
```
My ifconfig does not create the ipv6 tapping:
```
pi@bananapi:/etc/network$ ifconfig
br0       Link encap:Ethernet  HWaddr be:ed:5c:b8:63:a0  
          inet6 addr: fe80::bced:5cff:feb8:63a0/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:32 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:0 (0.0 B)  TX bytes:7632 (7.4 KiB)

eth0      Link encap:Ethernet  HWaddr 02:4e:06:c2:d9:95  
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:4 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:0 (0.0 B)  TX bytes:580 (580.0 B)
          Interrupt:117 Base address:0xc000 

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:16436  Metric:1
          RX packets:8 errors:0 dropped:0 overruns:0 frame:0
          TX packets:8 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:1104 (1.0 KiB)  TX bytes:1104 (1.0 KiB)

p2p0      Link encap:Ethernet  HWaddr 9a:3b:16:c7:d4:ba  
          inet6 addr: fe80::983b:16ff:fec7:d4ba/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

tunl0     Link encap:IPIP Tunnel  HWaddr   
          UP RUNNING NOARP  MTU:1480  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

wlan0     Link encap:Ethernet  HWaddr 98:3b:16:c7:d4:ba  
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:1109 errors:0 dropped:0 overruns:0 frame:0
          TX packets:295 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:158995 (155.2 KiB)  TX bytes:28931 (28.2 KiB)

```

Does this approach only work for eth0? (and not on wlan0)

**Comment by horopciuc** [comment]
Hello!
I'm trying to make 6lbr work on my RPi 3 in order to access the [bbbb:100] webpage. I have noticed that the 6lbr.conf is missing (maybe because of the last update), but instead there is a 6lbr.conf.rpi file. I wonder if the configuration should be done in this file, or I should create a new file named 6lbr.conf. Is there anyone who have succeeded in order to access the [bbbb::100] page and could guide me through this process?
I've tried countless pages, but unfortunately no success yet. 
Thanks!
